guides/developing_quickly_with_reload_mode.md

```diff
@@ -0,0 +1,91 @@
+## Developing Quickly with Auto-Reloading
+
+**Prerequisite**: This Guide requires you to know about Blocks. Make sure to [read the Guide to Blocks first](/introduction_to_blocks).
+
+<span id="advanced-features"></span>
+
+When you are building a Gradio demo, particularly out of Blocks, you may find it cumbersome to keep re-running your code to test your changes. 
+
+To make it faster and more convenient to write your code, we've made it easier to "reload" your Gradio apps instantly when you are developing in a **Python IDE** (like VS Code, Sublime Text, PyCharm, or so on) or generally running your Python code from the terminal. We've also developed an analogous "magic command" that allows you to re-run cells faster if you use **Jupyter Notebooks** (or any similar environment like Colab).
+
+This short Guide will cover both of these methods, so no matter how you write Python, you'll leave knowing how to build Gradio apps faster.
+
+### Python IDE Reload 🔥
+
+If you are building Gradio Blocks using a Python IDE, your file of code (let's name it `app.py`) probably looks something like this: 
+
+```python
+import gradio as gr
+
+with gr.Blocks() as demo:
+    gr.Markdown("Hello World")
+    with gr.Tabs():
+        with gr.TabItem("Tab 1")
+            ...
+
+if __name__ == "__main__":
+    demo.launch()
+```
+
+The problem is that anytime that you want to make a change to the layout or components, you have to close and  rerun your app by writing `python app.py`.
+
+Instead of doing this, you can run your code in **reload mode** by changing 1 word: `python` to `gradio`:
+
+In the terminal, run `gradio app.py`. That's it! 
+
+Now, you'll see that after you'll see something like this:
+
+```
+Launching in *reload mode* on: http://127.0.0.1:7860 (Press CTRL+C to quit)
+
+Watching...
+
+WARNING:  The --reload flag should not be used in production on Windows.
+```
+
+The important part here is the line that says `Watching...` What's happening here is that Gradio will be observing the directory where `app.py` file lives, and if the file changes, it will automatically rerun the file for you. So you can focus on writing your code, and your Gradio demo will refresh automatically 🥳
+
+⚠️ Now, there are a couple of important things to keep in mind when use the reload mode:
+
+* Because Gradio launches the demo for you in debug mode, you should *not* `launch()` your Gradio Interface or Gradio Blocks in the main block in your code. Instead, like we did above, you can `launch()` your demo within an `if __name__ == "__main__":` context, so that your code will continue to run normally if you decide to do `python app.py` as well:
```

Comment:
I think we can remove this part?

1. I don't think this is correct, because both `python` and `gradio` works when demo.launch is in `__main__` or outside `__main__`. Does it not work for you for some cases?
2. The message is not very clear

Reply:
Interesting, so I just tested it "kinda" works. For me, I  get the following message repeated twice:

```py
Launching in *reload mode* on: http://127.0.0.1:7861 (Press CTRL+C to quit)

Watching: 'c:\users\islam\dev\gradio-repos\gradio\gradio', 'C:\Users\islam\Desktop'

WARNING:  The --reload flag should not be used in production on Windows.
Running on local URL:  http://127.0.0.1:7862/
```

Basically what is happening is that two Gradio Interfaces are launched, but only the 2nd one is actually running in debug mode.

Reply:
For simpliclity, I'll remove this part of the Guide and hopefully my example is enough to show people what the "correct usage" should be